### DISCUSSION TITLE
MariaDB migration

### DISCUSSION THREAD

**Discussion by virgo77** [azerothcore/azerothcore-wotlk#11547] [General]
Hi there! I need some help about DB migration between 2 versions of MariaDB.

Actually, i run the ACORE DB on MariaDB 10.5.15 and all is fine.
As MariaDB had upgraded the last version 10.6 to 10.6.7, I would like to migrate my actual DB to the new MariaDB 10.6.7.

So, I export my DB from 10.5.15 to import to 10.6.7.

For AUTH, no pb.
But for CHARACTERS and WORLD, I got some errors, like this one for CHARACTERS:

> Query:
> CREATE TABLE `banned_addons` (
>   `Id` int(10) unsigned NOT NULL AUTO_INCREMENT,
>   `Name` varchar(255) NOT NULL,
>   `Version` varchar(255) NOT NULL DEFAULT '',
>   `Timestamp` timestamp NOT NULL DEFAULT current_timestamp() ON UPDATE current_timestamp(),
>   PRIMARY KEY (`Id`),
>   UNIQUE KEY `idx_name_ver` (`Name`,`Version`)
> ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4
> 
> Error occured at:2022-04-27 16:13:41
> Line no.:952
> Error Code: 1709 - Index column size too large. The maximum column size is 767 bytes

I searched the web for solutions, but none are working. Like, in my.ini, adding the lines:

> [mysqld]
> innodb_large_prefix=TRUE
> innodb_file_per_table=TRUE
> innodb_large_prefix=on
> innodb_file_format=Barracuda

I restarted the DB, but still no luck. I tried to change the CHARSET 'UTF8MB4' to 'UTF8', but it's not working. And, lastly I tried to search if my DB can run with "innodb_large_prefix" but I didn't find it. Maybe I forgot something? Any advices?

Thanks a lot :)

V.

**Comment by Kitzunu** [comment]
Just use MySQL :D

**Comment by Winfidonarleyan** [comment]
https://mariadb.com/kb/en/upgrading/
https://mariadb.com/kb/en/upgrading-mariadb-on-windows/
